release-24.1: clusterversion: set devBranch to false, version to beta.1

celiala opened the pull request:
**What this does:**

1. Sets `DevelopmentBranch = false`
2. Sets `version.txt` to `beta.1`

For [REL-710](https://cockroachlabs.atlassian.net/browse/REL-710)

**When to merge:**
This PR should not be merged until we intend to select the `v24.1.0-beta.1` release candidate; i.e. when release blockers are cleared, as per:
- https://cockroachlabs.atlassian.net/jira/dashboards/10216
- [beta.1 blockers tracking] https://cockroachlabs.atlassian.net/browse/REL-790


_______
Putting runbook here for the next time:

1. Create draft PR against master branch. 
  a. **NB: do not merge this onto master**; this is done solely to fix failing tests as needed (example from 24.1 https://github.com/cockroachdb/cockroach/pull/122044)
4. [cockroach_versions.go] set developmentBranch = false
5. search for `1000023` and replace with `23`, i.e. `1000023.2`, `1000023.1`
6. dev gen docs; dev gen bazel
7. dev test pkg/sql/catalog/bootstrap --rewrite -f TestInitialValuesToString 
8. replace hashes as directed until test passes in:
  a. pkg/sql/catalog/bootstrap/bootstrap_test.go (update value for `prevSystemHash`)
  b. pkg/sql/catalog/bootstrap/testdata/testdata (update value for `system hash` and `tenant hash`)
9. [this may need to happen _just_ on the PR for the release branch, as this is enabled on master] Pull in `.tcl` files from https://github.com/cockroachdb/cockroach/pull/112404 to disable multitenancy tests, which are enabled on master, but need to be disabled on release branches.
10. reach out to teams as needed to get help fixing any remaining failing tests
11. close the draft PR against master
12. once release branch is cut, cherry-pick this commit and create a PR against the release branch.
13. once release-blockers have cleared, the PR against the release branch can be merged.


_______
Release note: None
Epic: None
Release Justification: non-production / release-process-only change -- to prepare release branch for beta.1

celiala committed:
clusterversion: set devBranch to false, version to beta.1

Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/121290)
<!-- Reviewable:end -->

celiala commented:
🤔 this [error output for expected and found](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/14604700?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&expandBuildChangesSection=true&expandBuildTestsSection=true) looks the same?

pkg/storage/storage_test_/storage_test:TestWALFailover:
```
        expected:
        OK
        secondary = bar/auxiliary/wals-among-stores
        UnhealthyOperationLatencyThreshold() = (100ms,false)
        found:
        OK
        secondary = bar/auxiliary/wals-among-stores
        UnhealthyOperationLatencyThreshold() = (100ms,true)
```

celiala commented:
acceptance check is failing - looks like a flake / unrelated to this commit. re-running the failed job.
- GH Action Job:[ //pkg/acceptance:acceptance_test](action://pkg/acceptance:acceptance_test)
- Test: TestDockerCLI_test_demo/test_demo.tcl

Output:
```
=== RUN   TestDockerCLI_test_demo/test_demo.tcl
    util_docker.go:257: non-zero exit code: 1
    --- FAIL: TestDockerCLI_test_demo/test_demo.tcl (58.64s)
```
Seeing this in the logs:

```
expect: does "" (spawn_id exp3) match glob pattern "ERROR: column "foo" does not exist"? no
(echo 'select foo;'; echo 'select 1;') | /cockroach/cockroach sql

expect: does "(echo 'select foo;'; echo 'select 1;') | /cockroach/cockroach sql\r\n" (spawn_id exp3) match glob pattern "ERROR: column "foo" does not exist"? no
ERROR: column "foo" does not exist
SQLSTATE: 42703
```

celiala commented:
Pulling in these tcl files to disable multitenancy tests [1], which are enabled on master, but need to be disabled on release branches. This error surfaced as part of this PR setting developmentBranch=false.

[1] https://github.com/cockroachdb/cockroach/pull/112404

celiala committed:
demo: temporarily enable multitenancy

Temporarily enabling multitenancy to unblock acceptance tests.

Created a tracking issue for us to revert this once tcl
files have been updated as needed:

https://github.com/cockroachdb/cockroach/issues/122053

celiala commented:
[pulling in these tcl files](https://github.com/cockroachdb/cockroach/commit/8614bb791599102984e0728a16b3f06799d2bf4e) from https://github.com/cockroachdb/cockroach/pull/112404 did not fix the [acceptance tests](https://github.com/cockroachdb/cockroach/actions/runs/8606977230?pr=121290).

Instead, I'm going to temporarily set [demoCtx.Multitenant back to true](https://github.com/cockroachdb/cockroach/pull/121290/commits/7544fc668e0401ae7ab69cc929c43561a584ef89), and create a tracking issue to fix before we ship 24.1 https://github.com/cockroachdb/cockroach/issues/122053.

celiala commented:
We have zero 24.1 release-blockers 🎉 (https://go.crdb.dev/blockers/24.1):

I'm going to merge this PR, then build a beta release candidate

rail commented on pkg/cli/context.go:
```diff
@@ -24,7 +24,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/cli/clisqlexec"
 	"github.com/cockroachdb/cockroach/pkg/cli/clisqlshell"
 	"github.com/cockroachdb/cockroach/pkg/cli/democluster"
-	"github.com/cockroachdb/cockroach/pkg/clusterversion"
```
Looks like this breaks the build.

srosenberg replied:
yeah, the roachtest nightlies failed to build,

- [GCE](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/14807967?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=true)
- [AWS](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAwsBazel/14807968?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true)

celiala replied:
doh, sorry about that! 

thank you, David, for fixing the merge skew https://github.com/cockroachdb/cockroach/pull/122275
